.github/workflows/run_rc_validation_python_yaml.yml

```diff
@@ -0,0 +1,286 @@
+              type: fixed
+              size: 30s
+          options:
+            streaming: true
+          EOF
+          echo "YAML file created:"
+          cat ${{ env.YAML_PIPELINE_FILE }}
+        shell: bash
+
+      - name: Run YAML Pipeline (Dataflow Runner), Wait, Extract ID, Cleanup Submitter
+        id: submit_yaml_df
+        run: |
+          echo "Running YAML Pipeline with DataflowRunner in Background..."
+          source beam_env/bin/activate
+          python -m apache_beam.yaml.main \
+            --yaml_pipeline_file=${{ env.YAML_PIPELINE_FILE }} \
+            --runner DataflowRunner \
+            --region=${{ env.GCE_REGION }} \
+            --project=${{ env.GCP_PROJECT_ID }} \
+            --temp_location ${{ env.GCS_UNIQUE_FOLDER_PREFIX }}/temp \
+            --staging_location ${{ env.GCS_UNIQUE_FOLDER_PREFIX }}/staging \
+            > yaml_dataflow_submit.log 2>&1 &
+
+          YAML_DF_PID=$!
+          echo "YAML Pipeline (Dataflow Runner) submission process started in background with PID: ${YAML_DF_PID}"
+          echo ${YAML_DF_PID} > yaml_dataflow_submit.pid
+
+          echo "Waiting up to ${{ env.SUBMISSION_TIMEOUT_SECONDS }} seconds for Dataflow job submission process (PID: ${YAML_DF_PID}) to potentially complete..."
+          sleep ${{ env.SUBMISSION_TIMEOUT_SECONDS }}
+
+          echo "Proceeding with Job ID extraction..."
+          # Try extracting Job ID using common patterns from Dataflow submission logs
+          JOB_ID=$(grep -oP 'Dataflow Job ID: \K\S+' yaml_dataflow_submit.log || grep -oP "job_id='?\K[^' >]+" yaml_dataflow_submit.log || grep -oP "id: '?\"?\K[^'\" >]+" yaml_dataflow_submit.log | head -n 1)
+
+          if [[ -n "$JOB_ID" ]]; then
+            echo "Extracted YAML Dataflow Job ID: $JOB_ID"
+            echo "$JOB_ID" > yaml_dataflow_jobid.txt
+          else
+            echo "ERROR: Could not extract YAML Dataflow Job ID after ${{ env.SUBMISSION_TIMEOUT_SECONDS }}s wait. Log content:"
+            echo "--- YAML Dataflow submission log START ---"
+            cat yaml_dataflow_submit.log || echo "Log file not found."
+            echo "--- YAML Dataflow submission log END ---"
+            # Exit the step with failure if job ID is crucial and not found
+            exit 1
+          fi
+
+          # Check if the submission process is still running and kill it if necessary
+          if [ -f yaml_dataflow_submit.pid ] && ps -p $YAML_DF_PID > /dev/null; then
+             echo "Submission process (PID: $YAML_DF_PID) is still running after ${{ env.SUBMISSION_TIMEOUT_SECONDS }}s. Attempting to kill it."
+             kill -9 $YAML_DF_PID || echo "Failed to kill process $YAML_DF_PID."
+          else
+             echo "Submission process (PID: $YAML_DF_PID) has already finished or PID file is missing."
+          fi
+          # Clean up PID file regardless
+          if [ -f yaml_dataflow_submit.pid ]; then
+              rm yaml_dataflow_submit.pid
+          fi
```

Comment:
We can skip all of this, since it will get auto-cleaned up at the end

Reply:
yes but want to kill it since we no longer need to run this Python process after this step.